### PR TITLE
LPD-15996 Order notes fixed permission checking

### PR DESCRIPTION
**Pull request by gianmarcobrunialti**
https://liferay.atlassian.net/browse/LPD-15996

### CODE REVIEW

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderModelResourcePermissionLogic.java [hunk 1]
```diff
@@ -148,6 +148,10 @@ public Boolean contains(
 				CommerceOrderActionKeys.
 					MANAGE_COMMERCE_ORDER_RESTRICTED_NOTES)) {
 
+			if (_hasRoleAccountSupplier(permissionChecker, commerceOrder)) {
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderModelResourcePermissionLogic.java:151]: why this one?
supplier should already have the MANAGE_COMMERCE_ORDER_RESTRICTED_NOTES permission, we should only check that we are checking for an order the supplier has access to
I don't think this is right

Comment (gianmarcobrunialti) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderModelResourcePermissionLogic.java:151]: Depending on the context, the peformed check upon calling the `CommerceOrderNoteService` (`getCommerceOrderNotes`, `addCommerceOrderNote`, etc.) is solely based on the specific `MANAGE_COMMERCE_ORDER_NOTES` or `MANAGE_COMMERCE_ORDER_RESTRICTED_NOTES`.

This in turn checks the fact that the user belongs to accounts which contain `MANAGE_COMMERCE_ORDERS` (or being the owner of the note), and Supplier does not. The `_hasRoleAccountSupplier` in fact checks the allowance over orders on the channel the supplier is bound to, and wasn't covered in the case of notes, both for MVC actions and for POST/PATCH from `headless-commerce-admin-order`. We can take this on an offline call to make sure.

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderModelResourcePermissionLogic.java (hunk 1 above)
Comment (gianmarcobrunialti) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderModelResourcePermissionLogic.java]: @luca-pellizzon Changed with the completed logic and moved to `containsManageNotes`, could you please check? :)

**Review by luca-pellizzon — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
@gianmarcobrunialti please check the results of LocalFile.CPCommerceOrders#CanUserWithPermissionsAddCommerceOrderNotesViaAPI

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
@luca-pellizzon I've been checking that test ever since yesterday morning as it did seem a genuine failure, but with or without the change, running the test locally, the test keeps failing and investigating just that. In the history the test started failing just with this PR though. I'll keep you posted.

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by victoroscuro** [comment]
ci:forward:force

**Pull request closed without merging**